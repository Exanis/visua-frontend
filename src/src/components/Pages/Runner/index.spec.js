import React from 'react';
import {shallow} from 'enzyme';

import {Runner} from './index';

describe('<Runner />', () => {
    const defaultProps = {
        runner: {
        },
        classes: {},
        runnerActions: {
            getRunnerToken: () => false
        },
        intl: {
            formatMessage: (message) => message.id
        }
    };


    it('should render Runner properly', () => {
        const rendererComponent = shallow(<Runner {...defaultProps} />);

        expect(rendererComponent).toMatchSnapshot();
    });
});
