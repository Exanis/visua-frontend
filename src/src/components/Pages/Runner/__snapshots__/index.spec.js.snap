// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Runner /> should render Runner properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Runner
    classes={Object {}}
    intl={
        Object {
            "formatMessage": [Function],
          }
    }
    runner={Object {}}
    runnerActions={
        Object {
            "getRunnerToken": [Function],
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className={undefined}
>
          <WithStyles(Typography)
                    className={undefined}
                    variant="body1"
          >
                    <FormattedMessage
                              defaultMessage="To connect a runner, use this code as JOIN_TOKEN environment var"
                              id="runner.explain"
                              values={Object {}}
                    />
          </WithStyles(Typography)>
          <WithStyles(Typography)
                    className={undefined}
                    variant="headline"
          >
                    <p />
          </WithStyles(Typography)>
</div>,
        <WithStyles(ItemsListBase)
          count={undefined}
          itemToPrimary={[Function]}
          itemToSecondary={[Function]}
          items={undefined}
          loadItems={undefined}
          loaded={undefined}
          title="runner.title"
/>,
      ],
      "className": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <WithStyles(Typography)
              className={undefined}
              variant="body1"
>
              <FormattedMessage
                            defaultMessage="To connect a runner, use this code as JOIN_TOKEN environment var"
                            id="runner.explain"
                            values={Object {}}
              />
</WithStyles(Typography)>,
            <WithStyles(Typography)
              className={undefined}
              variant="headline"
>
              <p />
</WithStyles(Typography)>,
          ],
          "className": undefined,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <FormattedMessage
                defaultMessage="To connect a runner, use this code as JOIN_TOKEN environment var"
                id="runner.explain"
                values={Object {}}
/>,
              "className": undefined,
              "variant": "body1",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "defaultMessage": "To connect a runner, use this code as JOIN_TOKEN environment var",
                "id": "runner.explain",
                "values": Object {},
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <p />,
              "className": undefined,
              "variant": "headline",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "p",
            },
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "count": undefined,
          "itemToPrimary": [Function],
          "itemToSecondary": [Function],
          "items": undefined,
          "loadItems": undefined,
          "loaded": undefined,
          "title": "runner.title",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className={undefined}
>
            <WithStyles(Typography)
                        className={undefined}
                        variant="body1"
            >
                        <FormattedMessage
                                    defaultMessage="To connect a runner, use this code as JOIN_TOKEN environment var"
                                    id="runner.explain"
                                    values={Object {}}
                        />
            </WithStyles(Typography)>
            <WithStyles(Typography)
                        className={undefined}
                        variant="headline"
            >
                        <p />
            </WithStyles(Typography)>
</div>,
          <WithStyles(ItemsListBase)
            count={undefined}
            itemToPrimary={[Function]}
            itemToSecondary={[Function]}
            items={undefined}
            loadItems={undefined}
            loaded={undefined}
            title="runner.title"
/>,
        ],
        "className": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <WithStyles(Typography)
                className={undefined}
                variant="body1"
>
                <FormattedMessage
                                defaultMessage="To connect a runner, use this code as JOIN_TOKEN environment var"
                                id="runner.explain"
                                values={Object {}}
                />
</WithStyles(Typography)>,
              <WithStyles(Typography)
                className={undefined}
                variant="headline"
>
                <p />
</WithStyles(Typography)>,
            ],
            "className": undefined,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <FormattedMessage
                  defaultMessage="To connect a runner, use this code as JOIN_TOKEN environment var"
                  id="runner.explain"
                  values={Object {}}
/>,
                "className": undefined,
                "variant": "body1",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "defaultMessage": "To connect a runner, use this code as JOIN_TOKEN environment var",
                  "id": "runner.explain",
                  "values": Object {},
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <p />,
                "className": undefined,
                "variant": "headline",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": "p",
              },
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "count": undefined,
            "itemToPrimary": [Function],
            "itemToSecondary": [Function],
            "items": undefined,
            "loadItems": undefined,
            "loaded": undefined,
            "title": "runner.title",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
